src/include/ipc4/header.h

```diff
@@ -212,7 +212,6 @@ struct sof_ipc4_notify_module_data {
 	uint16_t module_id;
 	uint32_t event_id;
 	uint32_t event_data_size;
-	uint8_t event_data[];
```

Comment:
This is a good catch if we are using this structure within other structures. I would fix the beamformer usage here to use a pointer for this object.

Reply:
i.e.
```diff
diff --git a/src/audio/tdfb/tdfb_ipc4.c b/src/audio/tdfb/tdfb_ipc4.c
index 5360b1013..17c4ff144 100644
--- a/src/audio/tdfb/tdfb_ipc4.c
+++ b/src/audio/tdfb/tdfb_ipc4.c
@@ -28,7 +28,7 @@
 #include "tdfb_comp.h"
 
 struct tdfb_notification_payload {
-       struct sof_ipc4_notify_module_data module_data;
+       struct sof_ipc4_notify_module_data *module_data;
        struct sof_ipc4_control_msg_payload control_msg;
        struct sof_ipc4_ctrl_value_chan control_value; /* One channel value */
 };
```

Reply:
Also I don't think I can populate the IPC message data with the above suggestion where data would be a pointer to some other memory area, not part of the IPC message. I think there was a mistake in the structs design that this patch fixes. Though gcc did not complain about this, nor did xt-xcc, only xt-clang that caught it.

Reply:
Currently there is no such usage but not sure there ever won't be. So,  I think I'll change to construct the message header in heap and copy it to allocated message.

Reply:
Changed to draft, I had a stupid mistake in msg pointer with wrong parenthesis placement, so there was a crash in firmware in 2nd successive capture.